File: common/src/main/java/com/urop/common/Task.java
package com.urop.common;

public class Task {
    static int globalID = 0;
    final public String cmd;
    final public String id;
    public int[] iArrData;
    public byte[] bArrData;
    public String strData;
    public int waitCount; // working time


    public Task(String cmd, String id) {
        this.cmd = cmd;
        this.id = id;
    }

    public Task(String cmd) {
        this(cmd, Integer.toString(globalID++));
    }

    public Task() {
        this("NOP");
    }

    public static Task Message(String message) {
        return new Task("Message", message);
    }
}

File: common/src/main/java/com/urop/common/Task.kt
package com.urop.common


open class Task(@JvmField val id: String = (globalID++).toString()) {

    companion object {
        var globalID = 0
    }
}

open class SortTask(@JvmField val start: Int, @JvmField val end: Int) : Task(encodeID(start, end)) {
    lateinit var arr: IntArray

    @JvmField
    var waitCount = 0

    companion object {
        fun encodeID(start: Int, end: Int): String {
            return "[$start,$end]"
        }
    }
}


class NOPTask : Task()

class QSortTask(start: Int = 0, end: Int = 0) : SortTask(start, end)
class MSortTask(start: Int = 0, end: Int = 0) : SortTask(start, end)

class Message(val msg: String = "") : Task()
//class Profile(val profile: HashMap<String, Int>): Task(){}

typealias Chessboard = IntArray

class NQueenTask(
    val chessboard: Chessboard,
    @JvmField var remaining: Int,
    @JvmField val step: Int
) : Task() {
    lateinit var solution: Array<Chessboard>

    constructor(size: Int = 0, step: Int = 0) : this(IntArray(size), size, step)
}